File: PizzaTime/PizzaTime/ClientApp/src/app/models/pizza.ts
import { Ingredient } from "./ingredient";

export class Pizza {

  Id: string;
  Name: string;
  Price: number;
  Description: string;
  Total: number;
  ImageUrl: string;

  Ingredients: Ingredient[] = new Array<Ingredient>();

  constructor(name: string, price?: number, description?: string, ingredients?: Ingredient[], imageUrl?: string, id?:string) {
    this.Name = name;
    this.Price = price ? price : 0;
    this.Ingredients = ingredients ? ingredients : new Array<Ingredient>();
    this.Description = description ? description : "";
    this.ImageUrl = imageUrl ? imageUrl : "";
    this.Id = id ? id : "";
  }

  RecalculateTotal(): void {

    this.Total = this.Price;

    this.Ingredients.forEach(ingredient => {

      if (ingredient && ingredient.Selected)
        this.Total += ingredient.Price;

    });

  };

}

File: AdminUWP/Models/Pizza.cs
using AdminUWP.Model;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AdminUWP.Models
{
    public class Pizza
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public decimal Price { get; set; }

        public string Description { get; set; }

        public string ImageUrl { get; set; }

        public ICollection<Ingredient> Ingredients { get; set; }

        public Pizza()
        {
            Ingredients = new List<Ingredient>();
        }

    }
}
